### PR TITLE
What's New Regarding rules.py

### PR DESCRIPTION
**Pull request by LukeC92**

### CODE REVIEW

**Comment by dkillick** [comment]
You can follow the pattern used by @corinnebosley in #2819.

**Comment by dkillick** [comment]
@LukeC92 please add `[ci skip]` to your commit message.

**Comment by dkillick** [comment]
@corinnebosley we must document all breaking changes to public API here. So if it's public and we've removed it, we need to document it in this whatsnew entry.

File: docs/iris/src/whatsnew/contributions_v2.0.0/incompatiblechange_2017-Oct-18_rules-removal.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+The following classes and functions have been removed from :mod:`iris.fileformats.rules`:
```
Comment (dkillick) [inline docs/iris/src/whatsnew/contributions_v2.0.0/incompatiblechange_2017-Oct-18_rules-removal.txt]: This line needs to be prefixed with "*&nbsp;".

File: docs/iris/src/whatsnew/contributions_v2.0.0/incompatiblechange_2017-Oct-18_rules-removal.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+	* ```iris.fileformat.rules.log```
+	* ```iris.fileformat.rules.CMAttribute```
+	* ```iris.fileformat.rules.CMCustomAttribute```
+	* ```iris.fileformat.rules.CoordAndDims```
+	* ```iris.fileformat.rules.DebugString```
+	* ```iris.fileformat.rules.FunctionRule```
+	* ```iris.fileformat.rules.ProcedureRule```
+	* ```iris.fileformat.rules.Rule```
+	* ```iris.fileformat.rules.RulesContainer```
+	* ```iris.fileformat.rules.RuleResult```
```
Comment (dkillick) [inline docs/iris/src/whatsnew/contributions_v2.0.0/incompatiblechange_2017-Oct-18_rules-removal.txt]: You're missing the removal of the `legacy_custom_rules` keyword arg from the `Loader` class in this module. Can you add a new line to the end of this document:

```
* Removed deprecated keyword argument ```legacy_custom_rules``` from :class:`iris.fileformats.rules.Loader`.
```

**Comment by LukeC92** [comment]
As far as I'm this covers all the changes made in the last 6 months. This covers some or all of the changes in the following Pull Requests: #2806, #2801, #2782, #2791, #2795, #2796, #2794, #2786, #2778, #2777.

**Comment by dkillick** [comment]
Thanks @LukeC92.

**Pull request merged**